File: ts/274.ts
function hIndex(citations: number[]): number {
    citations.sort((a, b) => a - b)
    let h = 0;
    for (let i = 0; i < citations.length; i++ ) {
        const smaller = Math.min(citations[i], citations.length - i)
        h = Math.max(h, smaller)
    }

    return h
};

File: kt/274.kt
class Solution {
    fun hIndex(citations: IntArray): Int {
        citations.sort()
        var h = 0
        for (i in citations.indices) {
            val smaller = citations[i].coerceAtMost(citations.size - i)
            h = h.coerceAtLeast(smaller)
        }
        return h
    }
}